prime_factor/prime_factor.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+
+void generate_primes(int *ar, int *cnt) {
+    *cnt = 0;
+    for (int i = 2; i < 60000 && *cnt < 1000; ++i) {
```

Comment:
We can put an upper bound to the number of possible primes for a given number. That is why we can remove the constants and put in something more desirable.